odoo/tools/sql.py

```diff
@@ -63,6 +63,8 @@ class SQL:
             SQL.identifier(tablename),
             SQL("%s = %s", SQL.identifier(columnname), value),
         )
+        # when the first argument is a SQL node, the result is SQL().join of arguments
+        sql = SQL(SQL("SELECT *"), SQL(" FROM table WHERE x = "), 5)
```

Comment:
I disagree with this.  It is unnecessary and can be confusing.  If you want to concatenate `SQL` objects, use the proper API, i.e., `SQL().join(...)`.

Reply:
It's quite a common case to just concatenate strings and it avoids having a "%s%s..." string or intermediary objects.